frontend/src/views/Environment.tsx

```diff
@@ -9,8 +9,20 @@ import { ReactNode } from "react";
 import ReactMarkdown from 'react-markdown'
 import mermaid from 'mermaid'
 import TabControl from '../components/TabControl'
-const hostname = process.env.REACT_APP_API_HOST || ''
-const wsHostname = process.env.REACT_APP_WS_HOST || ''
+
+const protocol = window && window.location && window.location.protocol;
+const hostname = window && window.location && window.location.hostname;
+const port = window && window.location && window.location.port;
+
+var backendURL = protocol + "//" + hostname + ":" + port;
+var wsBackendURL = "ws://" + hostname + ":" + port;
+
+if (process.env.REACT_APP_API_HOST !== undefined) {
```

Comment:
This still is only evaluated by webpack at bundle-time. If that fits the use case, its fine.

Reply:
@PBug90 From my point of view it would also be ok to remove REACT_APP_API_HOST entirely, as in a real deployment scenario it might be hard to know the hostname upfront. However, that's how it was implemented and it would be still usable to use different location for backend and frontend. However, wondering, as discussed before, if this would be a real-world use-case anyhow.

Reply:
It is still very useful for development purposes, thats why I would like to keep it. And iirc it is part of the standard-react-workflow used by create-react-app aswell.